### ISSUE TITLE
Hypermorph Eager Execution

### ISSUE THREAD

**Issue by n-adriel** [voxelmorph/voxelmorph#641]
**Edit: Resolved issue!** Using eager mode with a custom loss layer and add_loss() in its call function to access hyperparameter values.

**Task** (what are you trying to do/register?)

I am training a HyperMorph model to register CT images. Currently I am tuning the lambda parameter to use with MSE and Grad losses. My end goal is to learn the effects of the lambda and gamma hyperparameters for the semi-supervised model as described in the hypermorph paper, but I am having issue running with 1 hyperparameter.

A similar issue has been reported previously (#498) but the answers have not helped me.

Essentially there are two separate issues, 1) models do not work with eager mode disabled; 2) hyperparameter cannot be updated in loss function during training with eager mode enabled. I am able to run with eager mode enabled or disabled if there is a solution for either one.

Tensorflow: 2.9.2
Keras: 2.9.0
Python: 3.8
Note: I am running on a cluster environment and am unable to update package versions.
